src/sagemaker/jumpstart/curated_hub/accessors/curated_hub_s3_accessor.py

```diff
@@ -0,0 +1,102 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
nonblocking/something to think about: should we put these new utilities in a subdirectory under `accessors`? idk if the answer is yes/no